experiment/secret-sync-controller/client/client.go

```diff
@@ -0,0 +1,226 @@
+package client
+
+// Package client implements client creation utilities.
+// Creates K8s clientset and Secret Manager client.
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"os"
+	"path/filepath"
+
+	"k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	types "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	prowkube "k8s.io/test-infra/prow/kube"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	secretmanagerpb "google.golang.org/genproto/googleapis/cloud/secretmanager/v1"
+)
+
+func NewK8sClientset() (*kubernetes.Interface, error) {
+	kubeconfig := filepath.Join(os.Getenv("HOME"), ".kube", "config")
+	clientset, err := prowkube.GetKubernetesClient("", kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+	return &clientset, nil
+}
+
+func NewSecretManagerClient(ctx context.Context) (*secretmanager.Client, error) {
+	client, err := secretmanager.NewClient(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return client, nil
+}
+
+// structs for client interface
+type Interface interface {
+	ValidateKubernetesNamespace(namespace string) error
+	ValidateKubernetesSecret(namespace, id string) error
+	CreateKubernetesNamespace(namespace string) error
+	GetKubernetesSecretValue(namespace, id, key string) ([]byte, error)
+	UpsertKubernetesSecret(namespace, id, key string, data []byte) error
+	CreateKubernetesSecret(namespace, id string) error
```

Comment:
I guess I'm not sure I see the value in supporting the concept of a secret that has no data, which I think is what this is here for?

Reply:
The reason I have this function is entirely for setting up the test cases.
I didn't remove it from the `client.Interface` interface because I thought it wasn't that destructive/dangerous.